src/main/scala/chiseltest/legacy/backends/verilator/PatchCoverageCpp.scala

```diff
@@ -24,14 +24,21 @@ object PatchCoverageCpp {
   }
 
   private def replaceCoverage(cppFile: Path, lines: Array[String]): Unit = {
-    // we add our code at the beginning of the coverage section
-    val coverageStart = findLine(CoverageStartNeedle, cppFile, lines)
-    lines(coverageStart) += "\n" + CustomCoverInsertCode + "\n"
+    val (verilatorMajor, verilatorMinor) = VerilatorCppHarnessGenerator.verilatorVersion
+    require(verilatorMajor == 4)
 
-    // then we replace the call
-    val call = findLine(CallNeedle, cppFile, lines)
-    val callLine = lines(call).replaceAllLiterally(CallNeedle, CallReplacement)
-    lines(call) = callLine
+    if (verilatorMinor < 200) {
+      // we add our code at the beginning of the coverage section
+      val coverageStart = findLine(CoverageStartNeedle, cppFile, lines)
+      lines(coverageStart) += "\n" + CustomCoverInsertCode + "\n"
+
+      // then we replace the call
+      val call = findLine(CallNeedle, cppFile, lines)
+      val callLine = lines(call).replaceAllLiterally(CallNeedle, CallReplacement)
+      lines(call) = callLine
+    } else {
+      // ???
```

Comment:
Could you please add a comment here that states that starting with Verilator 4.2 the whole patch coverage hack is no longer necessary?

Reply:
Sure, that's a good idea. I was thinking of figuring out the right patch for 4.200, but I don't think I have the time for it now.
What do you think about removing `else` and moving the version condition for patch all the way up before [VerilatorExecutive.scala#L249](https://github.com/ucb-bar/chisel-testers2/blob/b07c3650403a4d9e7091c7bde03bae3aa247d2d8/src/main/scala/chiseltest/legacy/backends/verilator/VerilatorExecutive.scala#L249)?

Reply:
That sounds good to me. In 1-2 years, once there is an Ubuntu LTS release that has Verilator `4.202+` we can the just delete the whole `PatchCoverageCpp.scala`.